magnolia-voting-app/src/main/java/com/example/votingapp/PollStatusValueProvider.java

```diff
@@ -0,0 +1,39 @@
+package com.example.votingapp;
+
+import java.util.Calendar;
+
+import javax.jcr.*;
+
+import com.vaadin.data.ValueProvider;
+
+/**
+ * PollStatusColumnDefinition
+ */
+public class PollStatusValueProvider implements ValueProvider<Node, String> {
+    
+    @Override
+    public String apply(Node source) {
+        try {
+            if (source.hasProperties()) {
+                Calendar today = Calendar.getInstance();
+                Calendar startDate = source.getProperty("startDate").getValue().getDate();
+                Calendar expiryDate = source.getProperty("expiryDate").getValue().getDate();
+
+                int today_startDate_comparison_value = today.compareTo(startDate);
+                int today_expiryDate_comparison_value = today.compareTo(expiryDate);
+
+                if (today_startDate_comparison_value < 0) {
+                    return "To-do";
+                }
+                if (today_startDate_comparison_value >= 0 && today_expiryDate_comparison_value < 0) {
+                    return "On-going";
+                } 
+                return "Done";
```

Comment:
Try to test and apply this block
if (today_startDate_comparison_value < 0) {
                    return "To-do";
                } else if (today_expiryDate_comparison_value < 0) {
                    return "On-going";
                } else {
                    return "Done";
                }

Reply:
Because the method will be terminated immediately when it hits a return statement, I did not use else keywords.